backend/team6/src/main/java/programmers/team6/domain/vacation/service/VacationService.java

```diff
@@ -1,5 +1,220 @@
+		// 사용자 존재 여부 확인
+		memberRepository.findById(memberId)
+			.orElseThrow(() -> new RuntimeException("멤버 정보를 찾을 수 없습니다."));
+
+		// 페이지 요청 객체 생성 (페이지 번호는 0부터 시작)
+		Pageable pageable = PageRequest.of(page, 20);
+
+		// 휴가 신청 내역 페이징 조회
+		Page<VacationRequest> vacationRequestsPage = vacationRequestRepository.findByRequesterIdWithPaging(memberId,
+			pageable);
+
+		// 엔티티를 DTO로 변환
+		List<VacationListResponseDto> content = vacationRequestsPage.getContent().stream()
+			.map(request -> {
+				// 결재 단계 정보 조회 (첫 번째 결재자 정보만 가져옴)
+				ApprovalStep approvalStep = approvalStepRepository.findFirstByVacationRequestOrderByStepAsc(request)
+					.orElseThrow(() -> new RuntimeException("결재 단계 정보를 찾을 수 없습니다."));
+
+				// 휴가 유형 이름 조회
+				String vacationTypeName = request.getType().getName();
+
+				return vacationMapper.toVacationRequestListResponseDto(
+					request,
+					vacationTypeName,
+					request.getStatus().name(),
+					approvalStep.getMember().getName()
+				);
+			})
+			.collect(Collectors.toList());
+
+		// 페이징 응답 DTO 생성
+		return vacationMapper.toVacationListResponsePageDto(vacationRequestsPage, content);
+	}
+
+	// 휴가 신청 수정
+	@Transactional
+	public VacationUpdateResponseDto updateVacationRequest(Long memberId, VacationUpdateRequestDto requestDto) {
+		// 요청자 확인
+		Member requester = memberRepository.findById(memberId)
+			.orElseThrow(() -> new RuntimeException("멤버 정보를 찾을 수 없습니다."));
+
+		// 휴가 신청 조회
+		VacationRequest vacationRequest = vacationRequestRepository.findById(requestDto.getRequestId())
+			.orElseThrow(() -> new RuntimeException("휴가 신청 정보를 찾을 수 없습니다."));
+
+		// 수정 권한 확인 (본인이 신청한 휴가만 수정 가능)
+		if (!vacationRequest.canUpdate(memberId)) {
+			throw new RuntimeException("수정 권한이 없습니다.");
+		}
+
+		// 휴가 유형 코드 조회
+		Code vacationType = codeRepository.findByGroupCodeAndCode("VACATION_TYPE", requestDto.getVacationType())
+			.orElseThrow(() -> new RuntimeException("잘못된 휴가 유형입니다."));
+
+		// 휴가 신청 수정
+		vacationRequest.update(requestDto.getFrom(), requestDto.getTo(), requestDto.getReason(), vacationType);
+
+		// 결재자 정보 조회
+		ApprovalStep approvalStep = approvalStepRepository.findFirstByVacationRequestOrderByStepAsc(vacationRequest)
+			.orElseThrow(() -> new RuntimeException("결재 단계 정보를 찾을 수 없습니다."));
+
+		// 응답 DTO 생성
+		return vacationMapper.toVacationUpdateResponseDto(
+			vacationRequest,
+			vacationType.getName(),
+			approvalStep.getMember().getName()
+		);
+	}
+
+	// 대기중인 휴가 신청 취소
+	@Transactional
+	public boolean cancelVacationRequest(Long memberId, Long requestId) {
+		// 취소 요청자 확인
+		Member requester = memberRepository.findById(memberId)
+			.orElseThrow(() -> new RuntimeException("멤버 정보를 찾을 수 없습니다."));
+
+		// 휴가 신청 조회
+		VacationRequest vacationRequest = vacationRequestRepository.findById(requestId)
+			.orElseThrow(() -> new RuntimeException("휴가 신청 정보를 찾을 수 없습니다."));
+
+		// 이미 취소된 상태인지 확인
+		if (vacationRequest.getStatus() == ApprovalStatus.CANCELED) {
+			throw new IllegalStateException("이미 취소된 휴가 신청입니다.");
+		}
```

Comment:
상태 검사를 엔티티안으로 넣는 것도 고려해보면 좋을 것 같아요!
```suggestion
		if (vacationRequest.isCanceled()) {
			throw new IllegalStateException("이미 취소된 휴가 신청입니다.");
		}
```
혹은
```suggestion
		if (vacationRequest.isSameStatus(ApprovalStatus.CANCELED)) {
			throw new IllegalStateException("이미 취소된 휴가 신청입니다.");
		}
```
위 코드와 같이 만드는 걸 고려해보면 어떨까요?

Reply:
엔티티안으로 상태 검사를 넣는게 더 좋아보이네요 ! 수정 완료 ~